sql/expression/distinct.go

```diff
@@ -31,16 +33,36 @@ func (de *DistinctExpression) seenValue(ctx *sql.Context, value interface{}) (bo
 		de.dispose = dispose
 	}
 
-	hash, err := hashstructure.Hash(value, nil)
+	// TODO: how to handle nil?
```

Comment:
docstring might be helpful, I'm assuming nil doesn't hash so we need to keep a variable for it? Should the close function set `seenNil` back to false in case the expression is reused?

Reply:
Yep `nil` can't be hashed; I definitely need to set `seenNil` back to false in `Dispose()` method. Will fix!